prawcore/auth.py

```diff
@@ -277,6 +282,129 @@ def revoke(self, only_access=False):
             self.refresh_token = None
 
 
+class _OAuth2ClientUserAuthApp(object):
+    def __init__(self, authenticator, scopes, implicit=False, duration="permanent") -> None:
+        super().__init__()
+        self.__finished = False
+        self.__state = str(uuid4())
```

Comment:
It doesn't appear that a state can be specified. The intended usage of the state is for verification that you created the authentication request. 

How would it work if multiple people were to start the flow at the same time?

Reply:
The state was supposed to be user-specifyable. I pushed commit e12df98 to fix this. Multiple people would not be able to authorize with the same host/port at the same time. They would have to specify custom local ports to achieve this. The lifespan of "auth app" is from when `authorize_local_server` is executed and until an auth-code / implicit grant is returned successfully.